### PR TITLE
Add a pipeline step for loading application data

### PR DESCRIPTION
**Pull request by snopoke**
This adds a step to pipelines that can pull an app JSON from the commcare API. This is for a specific bot currently but it should also pave the way for more future integrations.

In particular this adds a way to store auth configurations which can be reused.

### CODE REVIEW

File: apps/analysis/tasks.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def run_analysis(run_group_id: int):
     elif group.is_complete:
         return
 
-    with RunStatusContext(group):
+    with RunStatusContext(group, bubble_errors=False):
```
Comment (SmittieC) [inline apps/analysis/tasks.py:92]: Why this change?

Comment (snopoke) [reply apps/analysis/tasks.py:92]: This is the outermost context manager so if it bubbles errors it will cause the celery task to fail which isn't what we want.

**Review by SmittieC — APPROVED** [review]
Looks fine to me
